drivers/adc/adc_nrfx_saadc.c

```diff
@@ -198,39 +268,38 @@ static int set_oversampling(const struct adc_sequence *sequence,
 
 	switch (sequence->oversampling) {
 	case 0:
-		nrf_oversampling = NRF_SAADC_OVERSAMPLE_DISABLED;
+		*oversampling = NRF_SAADC_OVERSAMPLE_DISABLED;
 		break;
 	case 1:
-		nrf_oversampling = NRF_SAADC_OVERSAMPLE_2X;
+		*oversampling = NRF_SAADC_OVERSAMPLE_2X;
 		break;
 	case 2:
-		nrf_oversampling = NRF_SAADC_OVERSAMPLE_4X;
+		*oversampling = NRF_SAADC_OVERSAMPLE_4X;
 		break;
 	case 3:
-		nrf_oversampling = NRF_SAADC_OVERSAMPLE_8X;
+		*oversampling = NRF_SAADC_OVERSAMPLE_8X;
 		break;
 	case 4:
-		nrf_oversampling = NRF_SAADC_OVERSAMPLE_16X;
+		*oversampling = NRF_SAADC_OVERSAMPLE_16X;
 		break;
 	case 5:
-		nrf_oversampling = NRF_SAADC_OVERSAMPLE_32X;
+		*oversampling = NRF_SAADC_OVERSAMPLE_32X;
 		break;
 	case 6:
-		nrf_oversampling = NRF_SAADC_OVERSAMPLE_64X;
+		*oversampling = NRF_SAADC_OVERSAMPLE_64X;
 		break;
 	case 7:
-		nrf_oversampling = NRF_SAADC_OVERSAMPLE_128X;
+		*oversampling = NRF_SAADC_OVERSAMPLE_128X;
 		break;
 	case 8:
-		nrf_oversampling = NRF_SAADC_OVERSAMPLE_256X;
+		*oversampling = NRF_SAADC_OVERSAMPLE_256X;
 		break;
 	default:
 		LOG_ERR("Oversampling value %d is not valid",
-			    sequence->oversampling);
+			sequence->oversampling);
 		return -EINVAL;
 	}
 
-	nrf_saadc_oversample_set(NRF_SAADC, nrf_oversampling);
 	return 0;
 }
 
```

Comment:
comment regarding `check_buffer_size` below - shouldn't we change `active_channels` to `channels_cnt` ? `active_channels` suggests that this is a bitmask